src/NzbDrone.Core/Indexers/TorrentRssIndexer/TorrentRssSettingsDetector.cs

```diff
@@ -0,0 +1,223 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.IO;
+using System.Xml;
+using System.Xml.Linq;
+using NLog;
+using NzbDrone.Common.Extensions;
+using NzbDrone.Core.Indexers.Exceptions;
+using NzbDrone.Core.Parser.Model;
+
+namespace NzbDrone.Core.Indexers.TorrentRssIndexer
+{
+    public class TorrentRssSettingsDetector
+    {
+        protected readonly Logger _logger = LogManager.GetCurrentClassLogger();
+
+        protected TorrentRssIndexerSettings _settings;
+
+        protected Func<IndexerRequest, IndexerResponse> _fetchIndexerResponseFunc;
+
+        /// <summary>
+        /// Detect settings for Parser, based on URL
+        /// </summary>
+        /// <param name="settings">Indexer Settings to use for Parser</param>
+        /// <param name="fetchIndexerResponseFunc">Func to retrieve Feed</param>
+        /// <returns>Parsed Settings or <c>null</c></returns>
+        public TorrentRssIndexerParserSettings Detect(TorrentRssIndexerSettings settings, Func<IndexerRequest, IndexerResponse> fetchIndexerResponseFunc)
```

Comment:
Talked to markus about it. Better just have the Detector import IHttpClient via constructor (IoC as usual) and fetch the url directly like here: https://github.com/Sonarr/Sonarr/pull/193/files#diff-140395caeff547e2a3eed6e67b699e29R202
That way you don't need the Func in Detect(..) and GetParser(..) making those interfaces cleaner.